### PR TITLE
Adding new has_key_for? method to params

### LINKED ISSUE #72: Add method to params object to check if it has key for a Form
```crystal
if params.has_key_for?(SaveUser)
  SaveUser.new(params)
else
  SaveUser.new
end
```

https://gitter.im/luckyframework/Lobby?at=5c942f1a8aa66959b64d3efb

### PR DESCRIPTION
**Pull request by jwoertink**
Fixes #72
This is another in the https://github.com/luckyframework/avram/pull/369 series.

This doesn't really do much in Avram itself since most of the implementation comes from Lucky::Params.

### CODE REVIEW

File: src/avram/params.cr [hunk 1]
```diff
@@ -38,4 +38,8 @@ class Avram::Params
   def nested_file(key) : Hash(String, String)
     @hash
   end
+
+  def has_key_for?(operation : Avram::Operation.class | Avram::SaveOperation.class) : Bool
+    nested?(operation.param_key).any?
```
Comment (matthewmcgarvey) [inline src/avram/params.cr:43]: Seems like this implementation can go in `Avram::Paramable`?

Comment (jwoertink) [reply src/avram/params.cr:43]: I thought about that too. I actually wrote the code in Lucky first, then came and did this. I think I'll do that.

File: src/avram/paramable.cr [hunk 2]
```diff
@@ -3,4 +3,8 @@ module Avram::Paramable
   abstract def nested(key : String) : Hash(String, String)
   abstract def get?(key : String)
   abstract def get(key : String)
+
+  def has_key_for?(operation : Avram::Operation.class | Avram::SaveOperation.class) : Bool
+    nested?(operation.param_key).any?
```
Comment (jwoertink) [inline src/avram/paramable.cr:8]: Since Avram::Params are mainly used to circumvent some testing stuff, the actual specs for this will go in Lucky itself.

Comment (paulcsmith) [reply src/avram/paramable.cr:8]: I think specs should definitely be here so we know we don't break Avram. 

I think we can still test this by making a fake class that implements this `Paramable` and then tests that it returns the right thing. 

See example here: https://github.com/luckyframework/avram/blob/79668a93943c7706bb775c01b5fd34d02c342c3f/spec/operations/nested_save_operation_spec.cr#L17-L58

**Pull request merged**